### PR TITLE
Migrate to target zenoh dev/1.0.0

### PR DESCRIPTION
**Pull request by Charles-Schleich**
1. Bump `serde_json`
2. Use of new `Encoding` and `Value` structs
3. Use of `Properties` -> `Parameters` Change

### CODE REVIEW

**Review by DariusIMP — APPROVED** [review]

File: Cargo.toml [hunk 1]
```diff
@@ -58,24 +58,25 @@ lazy_static = "1.4.0"
...
-zenoh-buffers = { version = "0.11.0-dev", git = "https://github.com/eclipse-zenoh/zenoh.git", branch = "main" }
-zenoh-codec = { version = "0.11.0-dev", git = "https://github.com/eclipse-zenoh/zenoh.git", branch = "main" }
-zenoh-collections = { version = "0.11.0-dev", git = "https://github.com/eclipse-zenoh/zenoh.git", branch = "main" }
-zenoh-core = { version = "0.11.0-dev", git = "https://github.com/eclipse-zenoh/zenoh.git", branch = "main" }
-zenoh-keyexpr = { version = "0.11.0-dev", git = "https://github.com/eclipse-zenoh/zenoh.git", branch = "main" }
-zenoh-plugin-trait = { version = "0.11.0-dev", git = "https://github.com/eclipse-zenoh/zenoh.git", branch = "main" }
-zenoh-protocol = { version = "0.11.0-dev", git = "https://github.com/eclipse-zenoh/zenoh.git", branch = "main" }
-zenoh-util = { version = "0.11.0-dev", git = "https://github.com/eclipse-zenoh/zenoh.git", branch = "main" }
-zenoh_backend_traits = { version = "0.11.0-dev", git = "https://github.com/eclipse-zenoh/zenoh.git", branch = "main" }
+zenoh-buffers = { version = "0.11.0-dev", git = "https://github.com/eclipse-zenoh/zenoh.git", branch = "dev/1.0.0" }
```
Comment (Mallets) [inline Cargo.toml:71]: In `dev/1.0.0` you should not be forced to import anything else than `zenoh` crate. Could you try to remove all crates excepts for `zenoh_backend_traits` and `zenoh-plugin-trait`?

File: src/client.rs [hunk 2]
```diff
@@ -29,9 +29,11 @@ use aws_sdk_s3::types::{
 };
 use aws_sdk_s3::Client;
 use aws_smithy_runtime::client::http::hyper_014::HyperClientBuilder;
-use zenoh::value::Value;
+use zenoh::internal::Value;
 use zenoh::Result as ZResult;
 use zenoh_buffers::buffer::SplitBuffer;
+use zenoh_buffers::ZBuf;
+// use zenoh_buffers::buffer::SplitBuffer;
```
Comment (Mallets) [inline src/client.rs:36]: Remove dead code.

File: src/client.rs [hunk 3]
```diff
@@ -128,14 +130,14 @@ impl S3Client {
         value: Value,
         metadata: Option<HashMap<String, String>>,
     ) -> ZResult<PutObjectOutput> {
-        let body = ByteStream::from(value.payload.contiguous().to_vec());
+        let body = ByteStream::from(ZBuf::from(value.payload()).contiguous().to_vec());
```
Comment (Mallets) [inline src/client.rs:133]: Use `value.payload().into::<Vec<u8>>()` instead.

File: Cargo.lock
Comment (Mallets) [inline Cargo.lock:1]: The `Cargo.lock` should not be removed but rather updated.

**Review by Mallets — APPROVED** [review]

**Pull request merged**